lib/mix/tasks/phx.gen.channel.ex

```diff
@@ -42,12 +42,31 @@ defmodule Mix.Tasks.Phx.Gen.Channel do
       {:eex, "channel_test.exs", Path.join(test_prefix, "channels/#{binding[:path]}_channel_test.exs")},
     ]
 
-    Mix.shell().info """
+    user_socket_path = Mix.Phoenix.web_path(context_app, "channels/user_socket.ex")
 
-    Add the channel to your `#{Mix.Phoenix.web_path(context_app, "channels/user_socket.ex")}` handler, for example:
+    instructions =
+      if File.exists?(user_socket_path) do
+        """
 
-        channel "#{binding[:singular]}:lobby", #{binding[:module]}Channel
-    """
+        Add the channel to your `#{user_socket_path}` handler, for example:
+
+            channel "#{binding[:singular]}:lobby", #{binding[:module]}Channel
+        """
+      else
+        """
```

Comment:
Let's ask the user if he wants us to create it for him. If he doesn't, then we print this error message. In fact, there is one cool thing, if we create the socket for the user, we can automatically add the channel line for it. :)

Reply:
Good point! I added the feature to ask the user. Please check again :)